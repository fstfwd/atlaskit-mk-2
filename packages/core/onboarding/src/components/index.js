// @flow

import Modal from './Modal';
import Spotlight from './Spotlight';
import SpotlightManager from './SpotlightManager';
import SpotlightTarget from './SpotlightTarget';
import SpotlightTransition from './SpotlightTransition';

export {
  Modal,
  Spotlight,
  SpotlightManager,
  SpotlightTarget,
  SpotlightTransition,
};
