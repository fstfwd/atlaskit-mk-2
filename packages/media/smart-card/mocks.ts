import * as fm from 'fetch-mock';

const context = '';
const googleDefinitionId = 'google';
const trelloDefinitionId = 'trello';

const serviceAuth = {
  key: 'default',
  displayName: 'Google',
  url:
    'https://outbound-auth-flow.ap-southeast-2.dev.atl-paas.net/start?containerId=f4d9cdf9-9977-4c40-a4d2-968a4986ade0&serviceKey=default',
};

const generator = {
  name: 'Google Drive',
  icon: {
    url:
      'https://ssl.gstatic.com/docs/doclist/images/infinite_arrow_favicon_5.ico',
  },
};

const resolvedBody = (definitionId: string, name: string) => ({
  meta: {
    visibility: 'restricted',
    access: 'granted',
    auth: [serviceAuth],
    definitionId,
  },
  data: {
    '@context': context,
    generator,
    name,
  },
});

const notFoundBody = (definitionId: string) => ({
  meta: {
    visibility: 'not_found',
    access: 'forbidden',
    auth: [serviceAuth],
    definitionId,
  },
  data: {
    '@context': context,
    generator,
  },
});

const unauthorisedBody = (definitionId: string) => ({
  meta: {
    visibility: 'restricted',
    access: 'unauthorized',
    auth: [serviceAuth],
    definitionId,
  },
  data: {
    '@context': context,
    generator,
  },
});

const forbiddenBody = (definitionId: string) => ({
  meta: {
    visibility: 'restricted',
    access: 'forbidden',
    auth: [serviceAuth],
    definitionId,
  },
  data: {
    '@context': context,
    generator,
  },
});

const delayP = (n: number) => new Promise(res => setTimeout(res, n));

export const mockMultipleCards = () => {
  let c1 = 0;
  let c2 = 0;
  fm.mock('*', async (_, opts: any) => {
    await delayP(1000);

    const resourceUrl = JSON.parse(opts.body).resourceUrl;

    if (resourceUrl.startsWith('google')) {
      c1++;
      const step = Math.floor((c1 - 1) / 5);
      if (step === 0) {
        return;
      }
      if (step === 1) {
        return unauthorisedBody(googleDefinitionId);
      }
      if (step === 2) {
        return;
      }
      if (step === 3) {
        return forbiddenBody(googleDefinitionId);
      }
      return resolvedBody(googleDefinitionId, resourceUrl);
    }

    if (resourceUrl.startsWith('trello')) {
      switch (++c2) {
        case 1:
          return unauthorisedBody(trelloDefinitionId);
        case 2:
          return forbiddenBody(trelloDefinitionId);
        default:
          return resolvedBody(trelloDefinitionId, resourceUrl);
      }
    }

    throw new Error('Unkonws request type');
  });
};

export const mockSingleCardWorkflow = () => {
  let c = 0;
  fm.mock(
    'https://api-private.stg.atlassian.com/object-resolver/resolve',
    () => {
      return delayP(10000).then(() => {
        switch (++c) {
          case 1:
            return unauthorisedBody;
          case 2:
            return forbiddenBody;
          default:
            return resolvedBody;
        }
      });
    },
  );
};
