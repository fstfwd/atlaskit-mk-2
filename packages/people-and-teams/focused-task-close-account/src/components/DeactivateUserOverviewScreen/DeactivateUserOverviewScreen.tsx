import * as React from 'react';
import { FormattedMessage } from 'react-intl';

import InfoIcon from '@atlaskit/icon/glyph/info';

import { deactivateOverviewMessages } from '../../messages';
import StatefulInlineDialog from '../StatefulInlineDialog';
import UserInfo from '../UserInfo';
import { DeactivateUserOverviewScreenProps } from './types';
import * as Styled from './styled';

/**
 * Copy not final - ROCKET-1610
 * i18n yet to be applied - ROCKET-1610
 */

export class DeactivateUserOverviewScreen extends React.Component<
  DeactivateUserOverviewScreenProps
> {
  static defaultProps: Partial<DeactivateUserOverviewScreenProps> = {
    isCurrentUser: false,
  };

  selectAdminOrSelfCopy = (adminCopy, selfCopy) => {
    return this.props.isCurrentUser ? selfCopy : adminCopy;
  };

  render() {
    const { accessibleSites, user } = this.props;

    return (
      <Styled.Screen>
        <Styled.Title>
          <FormattedMessage {...deactivateOverviewMessages.heading} />
        </Styled.Title>
        <FormattedMessage
          tagName="p"
          {...this.selectAdminOrSelfCopy(
            deactivateOverviewMessages.paragraphAboutToDeactivateAdmin,
            deactivateOverviewMessages.paragraphAboutToDeactivateSelf,
          )}
        />
        <UserInfo user={user} />
        <FormattedMessage
          tagName="p"
          {...this.selectAdminOrSelfCopy(
            deactivateOverviewMessages.paragraphWhenDeactivateAccountAdmin,
            deactivateOverviewMessages.paragraphWhenDeactivateAccountSelf,
          )}
        />
        <ul>
          <li>
            <FormattedMessage
              {...this.selectAdminOrSelfCopy(
                deactivateOverviewMessages.paragraphLoseAccessAdmin,
                deactivateOverviewMessages.paragraphLoseAccessSelf,
              )}
              values={{ fullName: user.fullName }}
            />
            <Styled.AccessibleSitesList>
              {accessibleSites.map((url, idx) => <li key={idx}>{url}</li>)}
            </Styled.AccessibleSitesList>
            <FormattedMessage
              {...this.selectAdminOrSelfCopy(
                deactivateOverviewMessages.paragraphLoseAccessFootnoteAdmin,
                deactivateOverviewMessages.paragraphLoseAccessFootnoteSelf,
              )}
              tagName="small"
            />
          </li>
          <li>
            <FormattedMessage
              {...this.selectAdminOrSelfCopy(
                deactivateOverviewMessages.paragraphContentCreatedAdmin,
                deactivateOverviewMessages.paragraphContentCreatedSelf,
              )}
            />{' '}
            <Styled.InfoIconWrapper>
              <StatefulInlineDialog
                placement="auto-start"
                content={
                  <FormattedMessage
                    {...this.selectAdminOrSelfCopy(
                      deactivateOverviewMessages.inlineDialogContentCreatedAdmin,
                      deactivateOverviewMessages.inlineDialogContentCreatedSelf,
                    )}
                  />
                }
              >
                <InfoIcon label="" size="small" />
              </StatefulInlineDialog>
            </Styled.InfoIconWrapper>
          </li>
          <li>
            <FormattedMessage
              {...this.selectAdminOrSelfCopy(
                deactivateOverviewMessages.paragraphNoLongerBilledAdmin,
                deactivateOverviewMessages.paragraphNoLongerBilledSelf,
              )}
            />
          </li>
        </ul>
        <FormattedMessage
          tagName="p"
          {...this.selectAdminOrSelfCopy(
            deactivateOverviewMessages.paragraphReactivateAtAnyTimeAdmin,
            deactivateOverviewMessages.paragraphReactivateAtAnyTimeSelf,
          )}
        />
      </Styled.Screen>
    );
  }
}

export default DeactivateUserOverviewScreen;
