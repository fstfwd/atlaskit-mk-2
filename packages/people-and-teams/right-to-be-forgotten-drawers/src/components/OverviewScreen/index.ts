export * from './OverviewScreen';
export { default } from './OverviewScreen';
