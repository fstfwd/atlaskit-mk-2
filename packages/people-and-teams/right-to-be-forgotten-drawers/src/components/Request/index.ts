export * from './Request';
export { default } from './Request';
