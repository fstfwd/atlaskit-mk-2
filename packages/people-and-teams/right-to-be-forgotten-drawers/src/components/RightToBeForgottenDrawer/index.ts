export { default } from './RightToBeForgottenDrawer';
export * from './RightToBeForgottenDrawer';
