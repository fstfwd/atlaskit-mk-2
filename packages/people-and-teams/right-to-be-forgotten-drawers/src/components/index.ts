export {
  default as RightToBeForgottenDrawer,
} from './RightToBeForgottenDrawer';
