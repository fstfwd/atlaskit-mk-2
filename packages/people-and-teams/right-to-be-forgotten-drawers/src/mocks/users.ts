export const catherineHirons = {
  id: 'chirons',
  fullName: 'Catherine Hirons',
  email: 'catherine.hirons@acme.com',
};
