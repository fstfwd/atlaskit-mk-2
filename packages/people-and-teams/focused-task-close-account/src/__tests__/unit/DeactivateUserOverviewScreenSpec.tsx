import * as React from 'react';
import { shallow } from 'enzyme';
import { DeactivateUserOverviewScreen } from '../../components/DeactivateUserOverviewScreen';
import { catherineHirons } from '../../mocks/users';
import accessibleSites from '../../mocks/accessibleSites';

const defaultProps = {
  accessibleSites,
  isCurrentUser: false,
  user: catherineHirons,
  onNext: jest.fn(),
  onCancel: jest.fn(),
  isLoading: false,
  getAccessibleSites: jest.fn(),
};

const render = (props = {}) =>
  shallow(<DeactivateUserOverviewScreen {...defaultProps} {...props} />);

test('DeactivateUserOverviewScreen as Admin', () => {
  expect(render({ isCurrentUser: false })).toMatchSnapshot();
});

test('DeactivateUserOverviewScreen as Self', () => {
  expect(render({ isCurrentUser: true })).toMatchSnapshot();
});
