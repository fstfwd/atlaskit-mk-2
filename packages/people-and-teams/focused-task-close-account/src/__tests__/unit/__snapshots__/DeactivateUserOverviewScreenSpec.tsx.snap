// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeactivateUserOverviewScreen as Admin 1`] = `
<styled.div>
  <styled.div>
    <FormattedMessage
      defaultMessage="Deactivate account"
      description="Heading for the screen that explains what happens when a user's account is deactivated"
      id="focused-task-close-account.deactivate-account.overview.heading"
      values={Object {}}
    />
  </styled.div>
  <FormattedMessage
    defaultMessage="You're about to deactivate the user account of"
    description="A paragraph explaining that the admin is about to deactivate another user"
    id="focused-task-close-account.deactivate-account.overview.paragraph.about-to-delete.admin"
    tagName="p"
    values={Object {}}
  />
  <UserInfo
    user={
      Object {
        "avatarUrl": "",
        "email": "catherine.hirons@acme.com",
        "fullName": "Catherine Hirons",
      }
    }
  />
  <FormattedMessage
    defaultMessage="When you deactivate the account:"
    description="A paragraph introducing the impact of deactivating the account"
    id="focused-task-close-account.deactivate-account.overview.paragraph.when-deactivate-account"
    tagName="p"
    values={Object {}}
  />
  <ul>
    <li>
      <FormattedMessage
        defaultMessage="They will immediately lose access to all Atlassian services. {fullName} currently has access to:"
        description="A paragraph explaining that upon deletion the user will lose access to certain services. The second sentence is to begin a list of services the user-to-be-deleted can currently access. {fullName} is filled in by the web app."
        id="focused-task-close-account.deactivate-account.overview.paragraph.loseAccess.admin"
        values={
          Object {
            "fullName": "Catherine Hirons",
          }
        }
      />
      <styled.ul>
        <li
          key="0"
        >
          hello.atlassian.net
        </li>
        <li
          key="1"
        >
          acme.atlassian.net
        </li>
        <li
          key="2"
        >
          test.atlassian.net
        </li>
      </styled.ul>
      <FormattedMessage
        defaultMessage="They might also have access to other services, such as {atlassianCommunity}"
        description="A paragraph explaining that upon deletion the user may also lose access to services that weren't listed prior."
        id="focused-task-close-account.deactivate-account.overview.paragraph.loseAccess.footnote.admin"
        tagName="small"
        values={
          Object {
            "atlassianCommunity": "Atlassian Commnuity",
          }
        }
      />
    </li>
    <li>
      <FormattedMessage
        defaultMessage="Their personal data, such as their full name and email address, and content they've created will remain in Atlassian services."
        description="A paragraph explaining that content you have created will remain."
        id="focused-task-close-account.deactivate-account.overview.paragraph.content-created.admin"
        values={Object {}}
      />
       
      <styled.span>
        <StatefulInlineDialog
          content={
            <FormattedMessage
              defaultMessage="For example, pages, issues, and comments they've created in products."
              description="Examples of what constitutes as created content. Appears when the user hovers over the info icon"
              id="focused-task-close-account.deactivate-account.overview.inline-dialog.content-created.admin"
              values={Object {}}
            />
          }
          placement="auto-start"
        >
          <InfoIcon
            label=""
            size="small"
          />
        </StatefulInlineDialog>
      </styled.span>
    </li>
    <li>
      <FormattedMessage
        defaultMessage="We'll no longer bill you for them."
        description="A paragraph explaining what parts of the user's personal data will be deleted"
        id="focused-task-close-account.deactivate-account.overview.paragraph.no-longer-billed.admin"
        values={Object {}}
      />
    </li>
  </ul>
  <FormattedMessage
    defaultMessage="You can reactivate the account at any time"
    description="A paragraph explaining what parts of the user's personal data will be deleted"
    id="focused-task-close-account.deactivate-account.overview.paragraph.no-longer-billed.admin"
    tagName="p"
    values={Object {}}
  />
</styled.div>
`;

exports[`DeactivateUserOverviewScreen as Self 1`] = `
<styled.div>
  <styled.div>
    <FormattedMessage
      defaultMessage="Deactivate account"
      description="Heading for the screen that explains what happens when a user's account is deactivated"
      id="focused-task-close-account.deactivate-account.overview.heading"
      values={Object {}}
    />
  </styled.div>
  <FormattedMessage
    defaultMessage="You're about to deactivate your account:"
    description="A paragraph explaining that user is about to deactivate their own account"
    id="focused-task-close-account.deactivate-account.overview.paragraph.about-to-delete.self"
    tagName="p"
    values={Object {}}
  />
  <UserInfo
    user={
      Object {
        "avatarUrl": "",
        "email": "catherine.hirons@acme.com",
        "fullName": "Catherine Hirons",
      }
    }
  />
  <FormattedMessage
    defaultMessage="When you deactivate your account:"
    description="A paragraph introducing the impact of deactivating your account"
    id="focused-task-close-account.deactivate-account.overview.paragraph.when-deactivate-account"
    tagName="p"
    values={Object {}}
  />
  <ul>
    <li>
      <FormattedMessage
        defaultMessage="You will immediately lose access to all Atlassian services. You currently have access to:"
        description="A paragraph explaining that upon deletion the user will lose access to certain services. The second sentence is to begin a list of services the user-to-be-deleted can currently access."
        id="focused-task-close-account.deactivate-account.overview.paragraph.loseAccess.self"
        values={
          Object {
            "fullName": "Catherine Hirons",
          }
        }
      />
      <styled.ul>
        <li
          key="0"
        >
          hello.atlassian.net
        </li>
        <li
          key="1"
        >
          acme.atlassian.net
        </li>
        <li
          key="2"
        >
          test.atlassian.net
        </li>
      </styled.ul>
      <FormattedMessage
        defaultMessage="You might also have access to other services, such as {atlassianCommunity}"
        description="A paragraph explaining that upon deletion the user may also lose access to services that weren't listed prior."
        id="focused-task-close-account.deactivate-account.overview.paragraph.loseAccess.footnote.self"
        tagName="small"
        values={
          Object {
            "atlassianCommunity": "Atlassian Commnuity",
          }
        }
      />
    </li>
    <li>
      <FormattedMessage
        defaultMessage="Your personal data, such as your full name and email address, and content you've created will remain in Atlassian services."
        description="A paragraph explaining that content they have created will remain."
        id="focused-task-close-account.deactivate-account.overview.paragraph.content-created.self"
        values={Object {}}
      />
       
      <styled.span>
        <StatefulInlineDialog
          content={
            <FormattedMessage
              defaultMessage="For example, pages, issues, and comments you've created in products."
              description="Examples of what constitutes as created content. Appears when the user hovers over the info icon"
              id="focused-task-close-account.deactivate-account.overview.inline-dialog.content-created.self"
              values={Object {}}
            />
          }
          placement="auto-start"
        >
          <InfoIcon
            label=""
            size="small"
          />
        </StatefulInlineDialog>
      </styled.span>
    </li>
    <li>
      <FormattedMessage
        defaultMessage="You will no longer be included in the bill."
        description="A paragraph explaining what parts of the user's personal data will be deleted"
        id="focused-task-close-account.deactivate-account.overview.paragraph.no-longer-billed.self"
        values={Object {}}
      />
    </li>
  </ul>
  <FormattedMessage
    defaultMessage="You can reactivate the account at any time"
    description="A paragraph explaining that the account can be reactivated."
    id="focused-task-close-account.deactivate-account.overview.paragraph.no-longer-billed.self"
    tagName="p"
    values={Object {}}
  />
</styled.div>
`;
